lib/twilio-ruby/http/http_client.rb

```diff
@@ -17,29 +18,39 @@ def initialize(proxy_addr = nil, proxy_port = nil, proxy_user = nil, proxy_pass
         Faraday::Utils.default_params_encoder = Faraday::FlatParamsEncoder
       end
 
-      def request(host, port, method, url, params = {}, data = {}, headers = {}, auth = nil, timeout = nil)
-        @connection = Faraday.new(url: host + ':' + port.to_s, ssl: { verify: true }) do |f|
+      def _request(request)
+        @connection = Faraday.new(url: request.host + ':' + request.port.to_s, ssl: { verify: true }) do |f|
           f.request :url_encoded
           f.adapter @adapter
-          f.headers = headers
-          f.basic_auth(auth[0], auth[1])
+          f.headers = request.headers
+          f.basic_auth(request.auth[0], request.auth[1])
           if @proxy_addr
             f.proxy '#{@proxy_user}:#{@proxy_pass}@#{@proxy_addr}:#{@proxy_port}'
           end
-          f.options.open_timeout = timeout
-          f.options.timeout = timeout
+          f.options.open_timeout = request.timeout
+          f.options.timeout = request.timeout
         end
 
-        response = @connection.send(method.downcase.to_sym, url, method == 'GET' ? params : data)
+        @last_request = request
```

Comment:
can you set a `@last_response = nil` here so if theres an exception it wont contain the response from the call before?